fastestimator/dataset/extend_dataset.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import math
+
+from torch.utils.data import Dataset
+from fastestimator.dataset.batch_dataset import BatchDataset
+
+from fastestimator.util.traceability_util import traceable
+
+
+@traceable()
+class ExtendDataset(Dataset):
+    """ExtendDataset either extends or contracts the length of provided Dataset.
+
+    Args:
+        dataset: The Original dataset(s) which need expansion or contraction.
+        spoof_length: Length to which original dataset must be expanded or contracted to. (New desired length)
+    """
+    def __init__(self, dataset: Dataset, spoof_length: int) -> None:
+
+        self.dataset = dataset
```

Comment:
let's prevent ppl from nesting extendDatasets inside one another

Reply:
Its prevented in "checkinfo" method